models/stock_export.py

```diff
@@ -0,0 +1,282 @@
+            1) Goods In:  Reference, Part number, Pallet, Quantity;
+            2) Goods Out: Reference, Part number, Pallet, Quantity.
+        '''
+        if not self.date:
+            raise ValidationError(_("Date not found."))
+
+        self.field_data = False
+        self.field_name = False
+
+        Move = self.env['stock.move']
+        picking_type_in = self.env.ref('stock.picking_type_in')
+        picking_type_out = self.env.ref('stock.picking_type_out')
+
+        # @todo: check: usin union (no dups) instead of addition
+        picking_types = picking_type_in | picking_type_out
+
+        moves = Move.search([('state', '=', 'done'),
+                             ('picking_type_id', 'in', picking_types.ids),
+                             ('date', '>', self.date + " 00:00:00"),
+                             ('date', '<', self.date + " 23:59:59")])
+
+        in_moves = moves.filtered(lambda m: m.picking_type_id == picking_type_in)
+        out_moves = moves.filtered(lambda m: m.picking_type_id == picking_type_out)
+
+        _logger.info(_('Creating Excel file'))
+        file_name = 'warehouse_movemenet_{TIMESTAMP}.xls'.format(
+            TIMESTAMP=datetime.now())
+        wb = xlwt.Workbook()
+
+        for goods_moves, sheet_name in [(in_moves, "Goods In"),
+                                        (out_moves, "Goods Out")]:
+            sheet = wb.add_sheet(sheet_name)
+
+            for col, title in [(0, 'Reference'),
+                               (1, 'Part number'),
+                               (2, 'Pallet'),
+                               (3, 'Quantity')]:
+                sheet.write(0, col, title)
+
+            row = 0
+            for move in goods_moves:
+                for move_line in move.mapped('move_line_ids'):
+                    row += 1
+                    sheet.write(row, 0, move.picking_id.origin)
+                    sheet.write(row, 1, move.product_id.display_name)
+                    sheet.write(row, 2, move_line.result_package_id.name)
+                    sheet.write(row, 3, move_line.qty_done)
+
+        self.__write_workbook(wb, file_name, "Movement File")
+
+    @api.model
+    def __write_workbook(self, workbook, file_name, doc_title):
+        with closing(BytesIO()) as output:
+            workbook.save(output)
+            data = output.getvalue()
+
+        file_data = base64.b64encode(data)
+        self.file_name = file_name
+        self.file_data = file_data
+
+        attachment = self.env['ir.attachment'].create(
+            {'name': file_name,
+             'type': 'binary',
+             'datas': file_data,
+             'datas_fname': file_name})
+
+        _send_message_to_user(
+            self.env['res.users'],
+            subject="%s Ready" % doc_title,
+            body=_("%s %s is attached.") % (doc_title, file_name),
+            attachment=attachment)
+
+
+# @todo: (ale) consider moving this helper elsewhere
```

Comment:
Definitely move it elsewhere. I would suggest either adding to to the res.users model, or the mail.message model.